lib/graphql/query/variables.rb

```diff
@@ -34,16 +34,22 @@ def initialize(ctx, ast_variables, provided_variables)
               if validation_result.valid?
                 if value_was_provided
                   # Add the variable if a value was provided
-                  memo[variable_name] = if provided_value.nil?
+                  memo[variable_name] = if ctx.query.interpreter?
+                    provided_value
```

Comment:
In order to unify these codepaths and behaviors, we don't coerce these at boot anymore. We leave them until the runtime encounters them.